image/default_image_file.go

```diff
@@ -0,0 +1,279 @@
+	utils2 "github.com/alibaba/sealer/image/utils"
+	"github.com/alibaba/sealer/logger"
+	v1 "github.com/alibaba/sealer/types/api/v1"
+	"github.com/alibaba/sealer/utils"
+)
+
+type DefaultImageFileService struct {
+}
+
+func (d DefaultImageFileService) Load(imageSrc string) error {
+	return Decompress(imageSrc)
+}
+
+func (d DefaultImageFileService) Save(imageName string, imageTar string) error {
+	return Compress(imageName, imageTar)
+}
+
+func (d DefaultImageFileService) Merge(image *v1.Image) error {
+	panic("implement me")
+}
+
+func Compress(imageName string, imageTar string) error {
+	named, err := reference.ParseToNamed(imageName)
+	if err != nil {
+		return err
+	}
+
+	ima, err := utils2.GetImage(named.Raw())
+	if err != nil {
+		return err
+	}
+	if imageTar == "" {
+		imageTar = ima.Spec.ID + common.TarGzipSuffix
+	}
+	if _, err := os.Stat(imageTar); err == nil {
+		return fmt.Errorf("file %s exist", imageTar)
+	}
+	var layerDBCompress []string
+	var layerCompress []string
+	for _, layer := range ima.Spec.Layers {
+		if layer.Hash == "" {
+			continue
+		}
+		layerID := store.LayerID(digest.NewDigestFromEncoded(digest.SHA256, layer.Hash))
+		layerDir := filepath.Join(common.DefaultLayerDir, digest.Digest(layerID).Hex())
+		layerCompress = append(layerCompress, layerDir)
+		digs := digest.Digest(layerID)
+		subDir := filepath.Join(common.DefaultLayerDBDir, digs.Algorithm().String(), digs.Hex())
+		layerDBCompress = append(layerDBCompress, subDir)
+	}
+	compressFiles := append(append(layerDBCompress, layerCompress...), filepath.Join(common.DefaultImageMetaRootDir, ima.Spec.ID+common.YamlSuffix))
+
+	if filepath.IsAbs(imageTar) {
+		err = os.MkdirAll(filepath.Dir(imageTar), common.FileMode0644)
+		if err != nil {
+			return err
+		}
+	}
+	file, err := os.Create(imageTar)
+	defer func() {
+		if err != nil {
+			utils.CleanFile(file)
+		}
+		_ = file.Close()
+	}()
+
+	zr := gzip.NewWriter(file)
+	tw := tar.NewWriter(zr)
+	defer func() {
+		_ = tw.Close()
+		_ = zr.Close()
+	}()
+	for _, src := range compressFiles {
+		if len(src) == 0 {
+			return errors.New("[compress] source must be provided")
+		}
+
+		if !filepath.IsAbs(src) {
+			return errors.New("src should be absolute path")
+		}
+		var f os.FileInfo
+		var newFolder string
+		f, err = os.Stat(src)
+		if err != nil {
+			return err
+		}
+		if f.IsDir() {
+			newFolder = strings.TrimPrefix(src, common.DefaultImageDir+"/")
+		} else {
+			newFolder = strings.TrimPrefix(filepath.Dir(src), common.DefaultImageDir+"/")
+		}
+
+		//use existing file
+		src = strings.TrimSuffix(src, "/")
+		srcPrefix := filepath.ToSlash(src + "/")
+		err = filepath.Walk(src, func(file string, fi os.FileInfo, err error) error {
+			// generate tar header
+			header, walkErr := tar.FileInfoHeader(fi, file)
+			if walkErr != nil {
+				return err
+			}
+			if file != src {
+				absPath := filepath.ToSlash(file)
+				header.Name = filepath.Join(newFolder, strings.TrimPrefix(absPath, srcPrefix))
```

Comment:
this will give those compressed files a absolute path, which means those files need to be decompressed to original path. I think there is no need to add new folder.